compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/TypeParamResolver.java

```diff
@@ -215,15 +216,17 @@ public BType visit(BRecordType typeInSymbol, BType boundType) {
 
     @Override
     public BType visit(BTupleType typeInSymbol, BType boundType) {
-        List<BType> newTupleTypes = new ArrayList<>();
+        List<BTupleMember> newTupleTypes = new ArrayList<>();
 
-        List<BType> tupleTypes = typeInSymbol.tupleTypes;
+        List<BTupleMember> tupleTypes = typeInSymbol.memberTypes;
         boolean areAllSameType = true;
 
-        for (BType type : tupleTypes) {
-            BType newType = resolve(type, boundType);
-            areAllSameType &= newType == type;
-            newTupleTypes.add(newType);
+        for (BTupleMember type : tupleTypes) {
```

Comment:
```suggestion
        for (BTupleMember tupleMember : tupleMembers) {
```

Please check and introduce these changes wherever applicable.